packages/dds/tree/src/feature-libraries/chunked-forest/chunkTree.ts

```diff
@@ -226,75 +231,149 @@ export function makePolicy(policy?: Partial<ChunkPolicy>): ChunkPolicy {
 	return withDefaults;
 }
 
-export function shapesFromSchema(
-	schema: StoredSchemaCollection,
-	policy: FullSchemaPolicy,
-): Map<TreeNodeSchemaIdentifier, ShapeInfo> {
-	const shapes: Map<TreeNodeSchemaIdentifier, ShapeInfo> = new Map();
-	for (const identifier of schema.nodeSchema.keys()) {
-		tryShapeFromSchema(schema, policy, identifier, shapes);
-	}
-	return shapes;
+export interface ShapeFromSchemaParameters {
+	/**
+	 * The collection of stored schemas containing node and field definitions.
+	 */
+	readonly schema: StoredSchemaCollection;
+	/**
+	 * Policy from the app for interpreting the stored schema.
+	 */
+	readonly policy: FullSchemaPolicy;
+	/**
+	 * Policy function to determine if a field should be encoded incrementally.
+	 * Incrementally encoding requires the subtree to not start in the middle of a larger uniform chunk.
+	 * Thus returning true from this callback indicates that shapes should not be produced which could
+	 *contain the incremental portion as a part of a larger shape.
+	 */
+	readonly shouldEncodeIncrementally: IncrementalEncodingPolicy;
+	/**
+	 * A cache for shapes which may be read and/or updated. A new (or cleared) cache must be provided
+	 * if recalling this function after any of the parameters except in `ShapeFromSchemaParameters` have changed.
```

Comment:
"this function" isn't well defined here, so I think this should be rephrased:

```suggestion
	 * A cache for shapes which may be read and/or updated.
	 * As the shape is a function of the other members of `ShapeFromSchemaParameters`, it must be replaced or cleared if any of the properties other than this cache are modified.
```